### PR TITLE
SLES-15-010361 rule

### PR DESCRIPTION
**Pull request by rumch-se**
#### Description:

- _SLES-15-010361 rule_

#### Rationale:

- _Review required_

- Fixes # _Issue number here (e.g. #26) or remove this line if no issue exists._

### CODE REVIEW

File: linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_ownership_system_commands_dirs/rule.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+description: |-
+    System commands files are stored in the following directories by default:
+    <pre>/bin 
+    /sbin 
+    /usr/bin 
+    /usr/sbin 
+    /usr/local/bin 
+    /usr/local/sbin
+    </pre>
+    All files in these directories should be owned by the <tt>root</tt> user.
```
Comment (vojtapolasek) [inline linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_ownership_system_commands_dirs/rule.yml:16]:
```suggestion
    All files in these directories should be owned by the <tt>root</tt> group. 
```

File: linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_ownership_system_commands_dirs/rule.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    <pre>/bin 
+    /sbin 
+    /usr/bin 
+    /usr/sbin 
+    /usr/local/bin 
+    /usr/local/sbin
+    </pre>
+    All files in these directories should be owned by the <tt>root</tt> user. 
+    If the directory, or any file in these directories, is found to be owned 
+    by a user other than root correct its ownership with the following command:
```
Comment (vojtapolasek) [inline linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_ownership_system_commands_dirs/rule.yml:18]:
```suggestion
    by a group other than root correct its ownership with the following command:
```

File: linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_ownership_system_commands_dirs/rule.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    <pre>$ sudo chgrp root <i>FILE</i></pre>
+rationale: |-
```
Comment (vojtapolasek) [inline linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_ownership_system_commands_dirs/rule.yml:20]: blank line before rationale please

File: linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_ownership_system_commands_dirs/rule.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    If the operating system were to allow any user to make changes to software
```
Comment (vojtapolasek) [inline linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_ownership_system_commands_dirs/rule.yml:21]:
```suggestion
    If the operating system allows any user to make changes to software
```

File: linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_ownership_system_commands_dirs/rule.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+identifiers:
+    cce@sle15: CCE-85742-5
+
+references:
+    disa: CCI-001499
+    nist: CM-5(6),CM-5(6).1
+    stigid@sle15: SLES-15-010361
+    srg: SRG-OS-000259-GPOS-00100
+
+ocil_clause: 'any of these files are not owned by root'
```
Comment (vojtapolasek) [inline linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_ownership_system_commands_dirs/rule.yml:43]:
```suggestion
ocil_clause: 'any of these files are not owned by root group'
```

File: linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_ownership_system_commands_dirs/rule.yml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+ocil: |-
+    Sysem commands are stored in the following directories:
```
Comment (vojtapolasek) [inline linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_ownership_system_commands_dirs/rule.yml:46]:
```suggestion
    System commands are stored in the following directories:
```

File: linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_ownership_system_commands_dirs/rule.yml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    <pre>/bin 
+    /sbin 
+    /usr/bin 
+    /usr/sbin 
+    /usr/local/bin 
+    /usr/local/sbin</pre>
+    For each of these directories, run the following command to find files not
+    owned by root:
```
Comment (vojtapolasek) [inline linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_ownership_system_commands_dirs/rule.yml:54]:
```suggestion
    owned by root group:
```

File: linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_ownership_system_commands_dirs/ansible/shared.yml [hunk 8]
```diff
@@ -0,0 +1,12 @@
+# platform = multi_platform_sle
+# reboot = false
+# strategy = restrict
+# complexity = medium
+# disruption = medium
+
+- name: "Retrive the system command files and set their ownership to root"
```
Comment (vojtapolasek) [inline linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_ownership_system_commands_dirs/ansible/shared.yml:7]:
```suggestion
- name: "Retrieve the system command files and set their group ownership to root"
```

File: linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_ownership_system_commands_dirs/oval/shared.xml [hunk 9]
```diff
@@ -0,0 +1,28 @@
+<def-group>
+  <definition class="compliance" id="file_ownership_system_commands_dirs" version="1">
+    {{{ oval_metadata("
+        Checks that system commands in /bin /sbin /usr/bin /usr/sbin /usr/local/bin /usr/local/sbin 
+        are owned by root.
```
Comment (vojtapolasek) [inline linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_ownership_system_commands_dirs/oval/shared.xml:5]:
```suggestion
        are owned by root group.
```

File: linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_ownership_system_commands_dirs/oval/shared.xml [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    </criteria>
+  </definition>
+
+  <unix:file_test  check="all" check_existence="none_exist" comment="system commands are owned by root" id="test_ownership_system_commands_files" version="1">
+    <unix:object object_ref="object_file_ownership_system_commands_files" />
+  </unix:file_test>
+
+  <unix:file_object comment="system commands files" id="object_file_ownership_system_commands_files" version="1">
+    <!-- Check that system commands within directories /bin /sbin /usr/bin /usr/sbin /usr/local/bin /usr/local/sbin 
+         belong to user with uid 0 (root) -->
```
Comment (vojtapolasek) [inline linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_ownership_system_commands_dirs/oval/shared.xml:18]:
```suggestion
         belong to group with gid 0 (root) -->
```

**Review by vojtapolasek — CHANGES_REQUESTED** [review]
Hello, thank you for the rule. See comments please.
I have one request - if I understand it right, the rule checks for GROUP ownership, not USER ownership. If I am correct, please change the rule identifier to file_groupownership_system_commands_dirs.
Please reflect this change also in OVAL test / object / state names. Thank you.

**Comment by rumch-se** [comment]
Hello @vojtapolasek  - thank you very much for your proposals. I created a new commit 67f9df096 in which I integrated all changes proposed by you.

File: linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_groupownership_system_commands_dirs/rule.yml [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+    /usr/bin 
+    /usr/sbin 
+    /usr/local/bin 
+    /usr/local/sbin
+    </pre>
+    All files in these directories should be owned by the <tt>root</tt> group. 
+    If the directory, or any file in these directories, is found to be owned 
+    by a group other than root correct its ownership with the following command:
+    <pre>$ sudo chgrp root <i>FILE</i></pre>
+rationale: |-
```
Comment (vojtapolasek) [inline linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_groupownership_system_commands_dirs/rule.yml:20]: blank line here, please

**Comment by vojtapolasek** [comment]
Thank you for changes. Please could you change also test / object / state identifiers in the oval check? I recommend doing it now, cause if you later want to create another oval which checks for user ownership, there might be a clash of names.

**Comment by rumch-se** [comment]
@vojtapolasek - Hello - I have done the proposed changes.

**Pull request merged**